### ISSUE TITLE
Get all keys

### ISSUE THREAD

**Issue by IgnisDa** [sopherapps/scdb#10]
Thank you for this project. I have been using scdb in my [project](https://github.com/ignisda/ryot) with huge success!

I do not see a way to get all the keys in the database. Would it be possible to add a `get_all()` impl like [this](https://stackoverflow.com/a/8419509/11667450) SO answer?

**Comment by Tinitto** [comment]
Hey @IgnisDa .....hmmnn.... interesting feature request. It is not currently possible to get all keys.

Could you try to create it basing on the interface shown in the SO question? And, maybe, raise a PR and we have a look.

So something like:
```rust
let num_of_keys: usize = store.length;
let first_key: Option<String>= store.key(0);
```

**Comment by IgnisDa** [comment]
I can not find a `length` method on `Store` so the above code you posted is not possible.

**Comment by Tinitto** [comment]
Exactly. Both lines are not possible right now. The PR would be to make them possible.

**Comment by IgnisDa** [comment]
I'd say this would be pretty nice interface.

**Comment by IgnisDa** [comment]
@Tinitto Any updates on this?

**Comment by Tinitto** [comment]
@IgnisDa , My suggestion was you write the code and raise a pull request (PR) because the feature you want does not exist yet.

**Comment by IgnisDa** [comment]
Ah okay. Will try to do that. I have no experience with writing database systems though.

**Comment by IgnisDa** [comment]
@Tinitto `store.length` should return the number of non-expired keys, right?

EDIT: That would be confusing since an expired key could still have an index. I think it would be much better if we implement `get_all` instead.

**Comment by Tinitto** [comment]
Expired keys never get retrieved in any way. `get` and `search` treat them as nonexistent. I think `.length` and any other such method should treat them the same way i.e. nonexistent.

**Comment by IgnisDa** [comment]
What about the proposed `.key(n)` method?

**Comment by Tinitto** [comment]
That too must not 'see' the expired key-values.

We can look at key expiry as something like a user has set a key with a given value. After some time, that key should be automatically deleted. 

That means after that time `ttl`, it should not be found in the database in anyway.

How we implement it might include some optimizations that maybe leave that expired key-value on file but the user must never see it because to the user, it does not exist.
